mlflow/genai/agent_server/utils.py

```diff
@@ -0,0 +1,58 @@
+import logging
+import os
+import subprocess
+from contextvars import ContextVar
+
+import mlflow
+from mlflow.exceptions import MlflowException
+from mlflow.tracking.fluent import _get_experiment_id, _set_active_model
+
+# Context-isolated storage for request headers
+# ensuring thread-safe access across async execution contexts
+_request_headers: ContextVar[dict[str, str]] = ContextVar[dict[str, str]](
+    "request_headers", default={}
+)
+
+logger = logging.getLogger(__name__)
+
+
+def set_request_headers(headers: dict[str, str]) -> None:
+    """Set request headers in the current context (called by server)"""
+    _request_headers.set(headers)
+
+
+def get_request_headers() -> dict[str, str]:
+    """Get all request headers from the current context"""
+    return _request_headers.get()
+
+
+def setup_mlflow_git_based_version_tracking() -> None:
+    """Initialize MLflow tracking and set active model with git-based version tracking."""
+    experiment_id = _get_experiment_id()
+    if experiment_id is None:
+        raise MlflowException(
+            "You must set MLFLOW_EXPERIMENT_ID in your environment to enable MLflow git-based "
+            "version tracking and real time tracing."
+        )
+
+    mlflow.set_experiment(experiment_id=experiment_id)
```

Comment:
We don't need this, unless you want to allow users use `setup_mlflow_git_based_version_tracking(experiment_id)`